transport-classes-kqueue/src/main/java/io/netty/channel/kqueue/AbstractKQueueChannel.java

```diff
@@ -154,18 +160,27 @@ public boolean isOpen() {
     }
 
     @Override
-    protected void doDeregister() throws Exception {
+    protected void doDeregister(ChannelPromise promise) {
         // As unregisteredFilters() may have not been called because isOpen() returned false we just set both filters
         // to false to ensure a consistent state in all cases.
         // Make sure we unregister our filters from kqueue!
-        readFilter(false);
-        writeFilter(false);
+        try {
+            readFilter(false);
+        } catch (IOException ignore) {
```

Comment:
Hum... why are `readFilter` and `writeFilter` declared to throw `IOException`, though?